titus-common/src/main/java/com/netflix/titus/common/network/reverseproxy/grpc/ByteArrayMarshaller.java

```diff
@@ -0,0 +1,29 @@
+package com.netflix.titus.common.network.reverseproxy.grpc;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteStreams;
+import io.grpc.MethodDescriptor;
+
+class ByteArrayMarshaller implements MethodDescriptor.Marshaller<Object> {
```

Comment:
Take a look at ServerInterceptors#useInputStreamMessages.  You can avoid having to work with bytes here.

Reply:
Good idea. I will try that in the next iteration (it is still beta).